lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureAuthManagedIdentityConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.azure;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import jakarta.validation.constraints.AssertTrue;
+
+import java.util.Optional;
+
+public class AzureAuthManagedIdentityConfig
+{
+    private String clientId;
+    private String resourceId;
+
+    public Optional<String> getClientId()
+    {
+        return Optional.ofNullable(clientId);
+    }
+
+    @ConfigSecuritySensitive
+    @Config("azure.managed-identity.client-id")
+    public AzureAuthManagedIdentityConfig setClientId(String clientId)
+    {
+        this.clientId = clientId;
+        return this;
+    }
+
+    public Optional<String> getResourceId()
+    {
+        return Optional.ofNullable(resourceId);
+    }
+
+    @ConfigSecuritySensitive
+    @Config("azure.managed-identity.resource-id")
+    public AzureAuthManagedIdentityConfig setResourceId(String resourceId)
+    {
+        this.resourceId = resourceId;
+        return this;
+    }
+
+    @AssertTrue(message = "Exactly one of azure.managed-identity.client-id or azure.managed-identity.resource-id must be set")
```

Comment:
When `auth-type` is set to `DEFAULT` then
Either 1) Both the properties `client-id` and `resource-id` not set Or 2) Only one of them is set right?

So Here in the description kind of misleading where it is mentioned that for one of the property must be set.